lib/web/server.cpp

```diff
@@ -364,14 +380,19 @@ void Socialite::Web::Server::handleConnection(NetStream* stream) {
 		rHeader.relocate = false;
 		rHeader.allowCache = true;
 
+		// memset(&header, 0x0, sizeof(HttpHeader));
+		header.postMap.erase(header.postMap.begin(), header.postMap.end());
+
 		bool serverMatched = false;
 		string data;
 		int fileFd = 0;
 		string fullURI;
 		regex getValues("^([^\\?]+)\\?([^\\?]+)$");
 
 		readHeader(stream, &header, &error);
-		if(error) {
+		
+		// error reading headers
+		if(error != 0 || header.host == "" || header.URI == "") {
```

Comment:
Chrome was sometimes sending weird blank headers? Seems surprising but this fixed a frequent 400 bug. Maybe it has to do with the way it closes the connection.